onnx/test/numpy_helper_test.py

```diff
@@ -546,6 +546,25 @@ def test_float8_e5m2fnuz_negative_nan(self):
         back = numpy_helper.float8e4m3_to_float32(to, fn=True, uz=True)
         self.assertTrue(np.isnan(back))
 
+    def test_from_dict(self):
+        map_proto = numpy_helper.from_dict({0: np.array(0.1), 1: np.array(0.9)})
+        self.assertIsInstance(map_proto, MapProto)
+
+    def test_from_dict_no_np_array(self):
+        with self.assertRaises(AttributeError):
```

Comment:
Perhaps raise type error here?

Reply:
That would be ideal, but the AttributeError is thrown in `from_array`: 
`onnx/test/numpy_helper_test.py:557: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
onnx/numpy_helper.py:485: in from_dict
    value_seq = from_list(values)
onnx/numpy_helper.py:397: in from_list
    sequence.tensor_values.extend([from_array(tensor)])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

arr = 0.1, name = None

    def from_array(arr: np.ndarray, name: Optional[str] = None) -> TensorProto:
        """Converts a numpy array to a tensor def.
    
        Args:
            arr: a numpy array.
            name: (optional) the name of the tensor.
    
        Returns:
            TensorProto: the converted tensor def.
        """
        tensor = TensorProto()
>       tensor.dims.extend(arr.shape)
E       AttributeError: 'float' object has no attribute 'shape'
`

I could check the type more strongly in `from_array`, but I do wonder if we need that given the mypy type annotation already suggesting `arr` should be a `np.ndarray`.

Reply:
I added an explicit type check and raise in `from_array`, lmk what you think.